src/Chapter2.hs

```diff
@@ -858,8 +889,12 @@ and reverses it.
   function, but in this task, you need to implement it manually. No
   cheating!
 -}
-rewind = error "rewind: Not Implemented!"
-
+rewind :: [a] -> [a]
+rewind = helper []
+  where 
+    helper :: [a] -> [a] -> [a]
+    helper ans [] = ans
+    helper ans (x:xs) = helper (x:ans) xs
```

Comment:
Great and idiomatic solution! 👏🏻 The `helper` function is usually called `go` in Haskell in such cases 🙂 

You can read a bit more about the `go` pattern in here: https://kowainik.github.io/posts/haskell-mini-patterns#recursive-go

Reply:
Thanks this was fun. Looking forward to chapter 3